queries/useGroupInviteQuery.ts

```diff
@@ -8,7 +8,11 @@ export const useGroupInviteQuery = (groupInviteId: string) => {
   const account = useCurrentAccount() as string;
 
   return useQuery<GroupInvite | undefined>({
+    // I need to learn more about how query keys work to determine
+    // the significance of this query key with an account
     queryKey: groupInviteQueryKey(account, groupInviteId),
+    // does this need to be in useQuery for any
```

Comment:
Nope doesn't need to be

Think of useQuery and really all of the query client usage more as a network state wrapper.
The queryFn is just the actual function within the wrapper

I think the majority of XState usage would not actually interact with much of the state of react query other than maybe some prefetches

XState should be our necessary consistent state
React Query should be the helpful things to load later on

If we zoom out a little from this instance and instead look at initial app load
Xstate should handle:
1. Loading the client/checking if that is correct/usable
2. Loading our necessary information (Conversations, Groups, Consent)
3. Optionally it can tell our queries to prefetch some data (like fetching profiles of the first 20 items in our conversation list)

React Query:
React Query handles the actual decision of fetching from the network when needed

If XState was also in charge of fetching all profiles it can become pretty messy and slow

But if React Query handles it, from a DX experience it's a lot cleaner to say:
useProfileData and rely on the caching/persistence offered by the library

Now if we want to look at this use case, we _can_ use React Query just for initial loading of what the group invite network data is, but it's really not providing much other than syntactic sugar on top of useState/useEffect

So I think what makes sense is to remove useQuery entirely from this case and manage the one time load in XState

Reply:
@alexrisch Let me get a little bit more of the code done and we can review it and continue iterating for what should live where from there. It's a little too tricky without my end ready for review first. 

TLDR: 

`xstate` will DEFINITELY not reinvent the wheel that `react-query` has built, so querying/caching/optimistic updating logic will live in `react-query` and `xstate` will react to, observe, and trigger things in `react-query` land. 

Initially, I am going to keep `zustand` as a layer to share state as the transition to `xstate` completely as a state _management_ layer will be more lengthy and there will always be times where we may not want it just to share state in place A and place B. This is a conversation we will continue to revisit to make our workflows as ergonomic as possible. 

> So I think what makes sense is to remove useQuery entirely from this case and manage the one time load in XState

Agree here, although I'm also a huge fan of consistency. SHOULD our entire client -> network communication be done through `queryClient`? My initial gut says yes and that's how I'll build this, but we can always revisit. 

Having consistency to me is more valuable than saving a few keystrokes, and it will be a valuable low risk exercise to understand how to interact with network via `queryClient` in `xstate` within the Controlled Dependencies layer. 

Lots of words, code will help! Stay tuned 📺